h/services/user_update.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from sqlalchemy.exc import SQLAlchemyError
+
+from h.services.exceptions import ConflictError, ValidationError
+
+
+class UserUpdateService(object):
+    def __init__(self, session):
+        """
+        Create a new UserUpdateService
+
+        :param session: the SQLAlchemy session object
+        """
+        self.session = session
+
+    def update(self, user, **kwargs):
+        """
+        Update a user model with the args provided.
+
+        :arg user: the group to update
+        :type user: ~h.models.User
+
+        :raise ValidationError: if setting an attribute on the model raises :exc:`ValueError`
+                                or if ``authority`` is present in ``kwargs``
+
+        :rtype: ~h.models.User
+        """
+
+        # Much repurcussion if a user's authority is changed at this point.
+        # May wish to re-evaluate later if users need to be moved between
+        # authorities.
+        if "authority" in kwargs:
```

Comment:
Because changing the authority of a user would be so disruptive, and is not something we currently support at any layer, I've opted to raise here if it's present. We can certainly change this later.

Reply:
Makes sense. Moving a user account (and any related objects) between authorities is not something we have a need for.